### ISSUE TITLE
Improve and expand automated test suite

### ISSUE THREAD

**Issue by jsolbrig** [NRLMMD-GEOIPS/geoips#513]
# Requested Update

### Description
*  [ In 2-3 sentences briefly describe the required bug fix or feature with this issue. ]

### Background and Motivation
*  [ OPTIONAL: Provide additional detailed information and background related to this issue ]

### Alternative Solutions
*  [ OPTIONAL: If appropriate, list alternative fixes that could be considered]
  
### Environment
*  [ IF APPROPRIATE: Corresponding repository versions ]

### Code to demonstrate issue
*  [ IF APPROPRIATE: Can include steps here to identify/demonstrate bug fix or desired feature ]

#### Integration Testing
- [ ] Investigate methods to improve performance of integration tests. For example:
- [ ] Research and implement methods of parallelizing the tests either via internal parallelization or by running the individual tests in parallel.
- [ ] Research and develop methods of automatically producing and using smaller test datasets rather than complete datasets.
- [ ] Create a new tests plugin interface for GeoIPS.
- [ ] Convert current tests to plugins.
- [ ] Add code profiling to help identify regressions in efficiency.

#### Unit Testing
- [ ] Assess code coverage from integration tests and add tests as needed to ensure more complete coverage.
- [ ] Add documentation describing how to construct integration tests for new product types.
- [ ] Assess currently available unit testing tools to determine whether Pytest is the appropriate tool for GeoIPS.
- [ ] Research unit test structures and methods employed by other Python packages.
- [ ] Use lessons learned from other packages to formalize and document the structure and methods employed in the GeoIPS unit testing suite.
- [ ] Update the pull-request review process to require that unit tests are added for all new features.
- [ ] Identify and implement automated methods to ensure that unit tests are added when required. For example, require that test coverage doesn’t drop due to a new pull-request.
- [ ] While developing major new features like the order-based procflow and the CLI, refactor functions as needed to implement unit tests.

**Comment by jsolbrig** [comment]
Closing in favor of https://github.com/NRLMMD-GEOIPS/geoips/milestone/14
